src/main/kotlin/org/partiql/scribe/targets/spark/SparkCalls.kt

```diff
@@ -64,4 +69,80 @@ public open class SparkCalls(private val log: ProblemCallback) : SqlCalls() {
+     * We perform the conversion by a
+     *
+     * Notes:
+     *  > https://spark.apache.org/docs/2.3.0/api/sql/index.html#date_add
+     *  > https://spark.apache.org/docs/latest/api/java/org/apache/spark/sql/functions.html#date_add(org.apache.spark.sql.Column,int)
+     *  > https://spark.apache.org/docs/latest/api/java/org/apache/spark/sql/functions.html#add_months(org.apache.spark.sql.Column,int)
+     *  > The number of days to can be negative to subtract days
+     */
+    @OptIn(PartiQLValueExperimental::class)
+    override fun dateAdd(part: DatetimeField, args: SqlArgs): Expr {
+        var quantity = args[0].expr
+        var date = args[1].expr
+        var parts = arrayOfNulls<Expr>(7)
+        when (part) {
+            DatetimeField.YEAR -> parts[0] = quantity
+            DatetimeField.MONTH -> parts[1] = quantity
+            // skip weeks
+            DatetimeField.DAY -> parts[3] = quantity
+            DatetimeField.HOUR -> parts[4] = quantity
+            DatetimeField.MINUTE -> parts[5] = quantity
+            DatetimeField.SECOND -> parts[6] = quantity
+            else -> error("Unexpected datetime part `$part`")
+        }
+        val interval = exprCall(id("make_interval"), parts.map { it ?:  exprLit(int32Value(0)) })
+
+        // Add detailed warning about this translation.
+        val intervalString = parts.joinToString(
+            prefix = "make_interval(",
+            postfix = ")",
+            separator = ", "
+        ) {
+           if (it != null) "<quantity>" else "0"
+        }
+        log.warn("PartiQL `date_add($part, <quantity>, <date>)` was replaced by Spark `<date> + $intervalString`.")
+
+        return exprBinary(
+            op = Expr.Binary.Op.PLUS,
+            lhs = date,
+            rhs = interval,
+        )
+    }
+
+    /**
+     * PartiQL's date_diff accepts a part whereas Spark's datediff returns the difference in days.
+     *
+     * We extract the datetime part and perform the difference.
+     *
+     * Notes:
+     *  > https://github.com/partiql/partiql-lang-kotlin/wiki/Functions#date_diff----since-v010
+     *  > https://spark.apache.org/docs/latest/api/java/org/apache/spark/sql/functions.html#datediff(org.apache.spark.sql.Column,org.apache.spark.sql.Column)
+     *  > https://spark.apache.org/docs/2.3.0/api/sql/index.html#datediff
+     */
+    @OptIn(PartiQLValueExperimental::class)
+    override fun dateDiff(part: DatetimeField, args: SqlArgs): Expr {
+        val extract = when (part) {
+            DatetimeField.YEAR -> "year"
+            DatetimeField.MONTH -> "month"
+            DatetimeField.DAY -> "day"
+            DatetimeField.HOUR -> "hour"
+            DatetimeField.MINUTE -> "minute"
+            DatetimeField.SECOND -> "second"
+            else -> error("Unexpected datetime part `$part`")
+        }
+        val d1 = exprCall(id(extract), listOf(args[0].expr))
+        val d2 = exprCall(id(extract), listOf(args[1].expr))
+        return exprBinary(
```

Comment:
also could add a log info message about the translation we do for `DATE_DIFF`. RedshiftCalls does something similar -- https://github.com/partiql/partiql-scribe/blob/main/src/main/kotlin/org/partiql/scribe/targets/redshift/RedshiftCalls.kt#L54